src/CharsetDetector.cs

```diff
@@ -306,12 +337,12 @@ private bool RunProber(byte[] buf, int offset, int len, CharsetProber charsetPro
             return false;
         }
 
-        private void FindInputState(byte[] buf, int len)
+        private void FindInputState(byte[] buf, int offset, int len)
         {
-            for (int i = 0; i < len; i++)
+            for (int i = offset; i < len; i++)
             {
                 // other than 0xa0, if every other character is ascii, the page is ascii
-                if ((buf[i] & 0x80) != 0 && buf[i] != 0xA0)
+                if ((buf[offset + i] & 0x80) != 0 && buf[offset + i] != 0xA0)

```

Comment:
Could you clarify why this need to do _offset_ again?

Reply:
It was written under the same assumption with (new) line 371.

Reply:
But I-index already starts with offset

Reply:
Oh, I made a mistake. I will patch it ASAP.